src/main/java/com/revature/models/TripDTO.java

```diff
@@ -3,18 +3,33 @@
 import java.time.LocalDateTime;
 import java.util.List;
 
+import javax.validation.Valid;
+import javax.validation.constraints.Future;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
 @NoArgsConstructor @Data
 public class TripDTO {
 	private int tripId;
+    @NotNull
+	@Pattern(regexp = "[a-zA-Z0-9 ]+", message = "Trip names may only contain letters, numbers, and spaces")
 	private String name;
+    @NotNull
 	private User driver;
 	private List<User> riders;
+    @NotNull
 	private int availableSeats;
+    @Valid
+    @NotNull
 	private Address departure;
+    @Valid
+    @NotNull
 	private Address destination;
+    @NotNull
+    @Future
```

Comment:
Will _all_ trips have a future date? What if we are querying information about previously completed trips? Are trips always deleted once they are completed?

Reply:
No. For some reason I was only thinking of creating new trip validation, I'll remove.